### PR TITLE
added week7

### PR DESCRIPTION
**Pull request by zaidkazi3008**

### CODE REVIEW

File: Week-7/Ex-1/p1.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// Given a linked list of N nodes. The task is to reverse this list.
+
+//ex-1
+/*
+input: LinkedList: 1->2->3->4->5->6
+Output: 6 5 4 3 2 1
+*/
+
+
+
```
Comment (imsks) [inline Week-7/Ex-1/p1.js:12]: Why empty lines

File: Week-7/Ex-1/p1.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const SingleLinkedList = require('../utilities/linkedList1')
+
+
+
+const arr = [1,2,3,4,5,6]
+const ll = new SingleLinkedList(arr);
```
Comment (imsks) [inline Week-7/Ex-1/p1.js:18]: write complete name. What's ll?

File: Week-7/Ex-3/p3.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      this.next = next;
+    }
+}
+class SingleLinkedList {
+    constructor(value){
+      if(value instanceof Array){
+          this.size = value.length;
+          this.head = value.reverse().reduce((acc,curr) => {
+              if(acc == null){
+                  acc = new Node(curr);
```
Comment (imsks) [inline Week-7/Ex-3/p3.js:18]: Don't use curly braces if "If" statement is of one line

File: Week-7/Ex-3/p3.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      let array = [];
+      let currentList = this.head; 
+      while (currentList !== null) {
+          array.push(currentList.element);
+          currentList = currentList.next;
+      }
+      array.push()
+      console.log(array.join(" -> "));
+      return;
+    }
```
Comment (imsks) [inline Week-7/Ex-3/p3.js:65]: Give one line space after your function block ends

File: Week-7/utilities/linkedList1.js [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+      while (current.next != null) {
+        current = current.next;
+      }
+
+      // append node
+      current.next = node;
+    }
+    this.size++;
+  }
+  insertAt(element, index) {
```
Comment (imsks) [inline Week-7/utilities/linkedList1.js:53]: use full names as insertAtIndex

**Review by imsks — APPROVED** [review]

File: Week-7/Ex-1/p1.js (hunk 1 above)
Comment (zaidkazi3008) [reply Week-7/Ex-1/p1.js:12]: sorry sir next time i will see this.

File: Week-7/Ex-1/p1.js (hunk 2 above)
Comment (zaidkazi3008) [reply Week-7/Ex-1/p1.js:18]: okay.

File: Week-7/Ex-3/p3.js (hunk 3 above)
Comment (zaidkazi3008) [reply Week-7/Ex-3/p3.js:18]: surely.

**Pull request merged**